modules/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/MockHttpProxyServer.java

```diff
@@ -26,19 +27,16 @@
  */
 abstract class MockHttpProxyServer implements Closeable {
 
-    private final HttpServer httpServer;
+    private HttpServer httpServer;
```

Comment:
Can we make this field volatile to avoid have to think about visibility? Technically we restart the server on a generic thread and shutdown at test cleanup time in a different thread.